pages/notifications/manage-notifications/index.tsx

```diff
@@ -334,6 +335,17 @@ const ManageNotifications = (props) => {
   const shouldRenderMigrationBanner =
     props.migrationBannerProps.migrationType && hasMigrationStatus;
 
+  //The subscription-notification alert rendered by "urlAction" contains the same content as the successful migration banners.
+  //this variable hides the alert when one of the migrations is successful and neither of them have failed
+  const hideAlertConfirmationMessage =
+    shouldRenderMigrationBanner &&
+    props.migrationBannerProps.migrationType ===
+      LOGIN_NOTICE_TYPES.SUBSCRIPTION_NOTIFICATIONS &&
+    (props.migrationBannerProps.applyMigrationStatus === 'SUCCEEDED' ||
+      props.migrationBannerProps.findMigrationStatus === 'SUCCEEDED') &&
+    props.migrationBannerProps.applyMigrationStatus !== 'FAILED' &&
+    props.migrationBannerProps.findMigrationStatus !== 'FAILED';
```

Comment:
Can we refactor this into a function with, I think they're called, guard clauses. I'm sure it works but its a nightmare understanding the logic here

Reply:
Do you mean optional chaining? the migrationBannerProps here are always going to be defined - we assign defaults above

Happy to extract to a fn

Reply:
cleaned that up a bit